spec/till_spec.rb

```diff
@@ -0,0 +1,142 @@
+
+  describe '#total' do
+
+    it 'calculates the amount due' do
+      expect(till.total(order)).to eq( { amount_due: 15.48 } )
+    end
+
+    context 'when amount discount is added' do
+
+      it 'calculates amount due - discount' do
+        till.add_discount(amount_discount)
+        expect(till.total(order)).to eq( { amount_due: 13.93 } )
+      end
+
+      it 'shows discount on receipt' do
+        till.add_discount(amount_discount)
+        till.total(order)
+        expect(till.take_payment(20.00)[:discount]).to eq(1.55)
+      end
+
+    end
+
+    context 'when line discount is added' do
+
+      it 'calculates amount due - discount' do
+        till.add_discount(line_discount)
+        expect(till.total(order)).to eq( { amount_due: 15.00 } )
+      end
+
+      it 'shows discount on receipt' do
+        till.add_discount(line_discount)
+        till.total(order)
+        expect(till.take_payment(20.00)[:discount]).to eq(0.48)
+      end
+
+    end
+
+  end
+
+  describe '#take_payment' do
+
+    before do
+      till.total(order)
+    end
+
+    it 'produces receipt with time' do
+      Timecop.freeze(Time.local(2015, 10, 10, 8, 0))
+      expect(till.take_payment(15.48)[:time]).to eq("2015.10.10 08:00:00")
+    end
+
+    it 'produces receipt with cafe name' do
+      expect(till.take_payment(15.48)[:name]).to eq("The Coffee Connection")
+    end
+
+    it 'produces receipt with table number' do
+      expect(till.take_payment(15.48)[:table]).to eq(1)
+    end
+
+    it 'produces receipt with address' do
+      expect(till.take_payment(15.48)[:address]).to eq("123 Lakeside Way")
+    end
+
+    it 'produces receipt with phone number' do
+      expect(till.take_payment(15.48)[:phone]).to eq("16503600708")
+    end
+
+    it 'produces receipt with customer name' do
+      expect(till.take_payment(15.48)[:customers]).to eq(["Jane", "John"])
+    end
+
+    it 'produces receipt with each items quanitity and cost' do
+      expect(till.take_payment(15.48)[:items]).to eq([ { item: "Cafe Latte",
+                                                            quantity: 2,
+                                                            price: 4.75
+                                                           },
+                                                           {
+                                                            item: "Flat White",
+                                                            quantity: 1,
+                                                            price: 4.75 } ])
+    end
+
+    it 'produces receipt with subtotal'  do
+      expect(till.take_payment(15.48)[:subtotal]).to eq(14.25)
+    end
+
+    it 'produces receipt with total cost' do
+      expect(till.take_payment(15.48)[:total]).to eq(15.48)
+    end
+
+    it 'produces receipt with tax amount' do
+      expect(till.take_payment(15.48)[:tax]).to eq(1.23)
+    end
+
+    it'produces receipt with cash paid' do
```

Comment:
Lint/SpaceBeforeFirstArg: Put space between the method name and the first argument.